src/domain.ts

```diff
@@ -62,13 +62,30 @@ interface DocMetaData {
   character: number;
 }
 
-interface TestMetaData {
+interface TestMetadataKwArgs {
```

Comment:
I think you can do this through a union type and create multiple types per variant

Reply:
yes right, I did that in UI code, but didnt check this. Will fix this

Reply:
fixed